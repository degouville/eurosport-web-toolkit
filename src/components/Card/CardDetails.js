import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as colors from '../../colors';
import ProgramDetails, { StyledIconWrapper, StyledSeparator, StyledText } from '../../elements/ProgramDetails';
import { medium } from '../../breakpoints';

const StyledContent = styled.div`
  padding: 16px 16px 8px;
  line-height: 1.3;
  flex: 1 auto;
  display: flex;
  flex-direction: column;
  width: auto;
`;

const StyledCategory = styled.div`
  font-size: 12px;
  font-weight: bold;
  text-transform: uppercase;
  color: ${colors.malibu};
  letter-spacing: 1px;
  margin: 0 0 10px;
`;

const StyledFooterContentWrapper = styled.div`
  display: block;
  width: 100%;
`;

const StyledFooter = styled.div`
  flex: 1 auto;
  display: flex;
  align-items: flex-end;
  font-size: 14px;
`;

const StyledTitle = styled.div`
  font-size: 14px;
  font-weight: bold;
  color: ${colors.athensGray};
  margin: 0 0 10px;
`;

const StyledDescription = styled.div`
  color: ${colors.manatee};
  font-size: 12px;
  line-height: 20px;
  margin: 0 0 10px;
`;

const StyledProgramDetails = styled(ProgramDetails)`
  flex-grow: 0;
  font-size: 12px;
  line-height: 1.3;

  ${StyledIconWrapper} {
    height: 15px;
    padding: 8px 0;
  }

  ${StyledText} {
    padding: 8px 8px 8px 0;
  }

  ${StyledSeparator} {
    margin: 0 8px 0 10px;
  }

  ${medium(css`
    font-size: 12px;
    line-height: 1.3;
  `)};
`;

const CardDetails = ({ card, icon, ...props }) => {
  const { category, title, description, timestamp, channel = null } = card;
  return (
    <StyledContent {...props}>
      <StyledCategory>{category}</StyledCategory>
      <StyledTitle>{title}</StyledTitle>
      {description && <StyledDescription>{description}</StyledDescription>}
      <StyledFooter>
        <StyledFooterContentWrapper>
          <StyledProgramDetails textDetail={timestamp} callsign={channel} customIcon={icon} />
        </StyledFooterContentWrapper>
      </StyledFooter>
    </StyledContent>
  );
};

CardDetails.propTypes = {
  card: PropTypes.shape({
    category: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    timestamp: PropTypes.string,
    channel: PropTypes.string,
  }).isRequired,
  icon: PropTypes.node,
};

CardDetails.defaultProps = {
  icon: null,
};

CardDetails.displayName = 'Cards.Content';

export default CardDetails;
