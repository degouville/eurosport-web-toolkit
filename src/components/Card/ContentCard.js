import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { Card, Header, Content, Footer } from './base';
import UppercaseHeading from '../../elements/UppercaseHeading';
import Text from '../../elements/Text';
import Icon from '../../elements/Icon';
import { colors } from '../..';

const StyledContentTitle = styled(Text)`
  font-weight: bold;
  color: ${colors.whiteLilac};
  display: block;
`;

const StyledContentDescription = styled(Text)`
  color: ${colors.whiteLilac};
  display: block;
`;

export const StyledLiveLabel = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0.2em 0.4em;
  border-radius: 2px;
  background-color: #e00034;
  color: white;
  font-weight: bold;
  font-size: 15px;
  position: absolute;
  top: 10px;
  right: 10px;
`;

export const StyledPlayIcon = styled(Icon)`
  position: absolute;
  bottom: 17px;
  left: 16px;
`;

const StyledImage = styled('img')`
  width: 100%;
`;

const StyledTimeStamp = styled(Text)`
  padding: 8px 0;
`;

const StyledCardFooter = styled(Footer)`
  * :nth-child(2) {
    margin-left: 1em;
    padding-left: 1em;
    border-left: 1px solid grey;
    padding: 8px;
  }
`;

const ContentCard = ({ card, type }) => {
  const { img, category, title, description, timestamp, channel } = card;
  const isLive = type === 'live';
  const isPlayable = isLive || type === 'vod';

  return (
    <Card>
      <Header>
        <StyledImage src={img} alt={description} />
        {isPlayable && <StyledPlayIcon type="play" height="60" />}
        {isLive && <StyledLiveLabel>Live</StyledLiveLabel>}
      </Header>
      <Content>
        <UppercaseHeading as="h2">{category}</UppercaseHeading>
        <StyledContentTitle>{title}</StyledContentTitle>
        {description && <StyledContentDescription>{description}</StyledContentDescription>}
        <StyledCardFooter>
          {isLive && <Icon height="15" type={channel} />}
          <StyledTimeStamp>{timestamp}</StyledTimeStamp>
        </StyledCardFooter>
      </Content>
    </Card>
  );
};

ContentCard.propTypes = {
  card: PropTypes.shape({
    img: PropTypes.string.isRequired,
    category: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string,
    timestamp: PropTypes.string,
    channel: PropTypes.string,
  }).isRequired,
  type: PropTypes.oneOf(['vod','article','live'])
};

export default ContentCard;
