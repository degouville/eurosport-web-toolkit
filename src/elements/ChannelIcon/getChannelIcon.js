import { ReactComponent as EComponent } from '../../assets/channels/E.svg';
import { ReactComponent as E1Component } from '../../assets/channels/E1.svg';
import { ReactComponent as E2Component } from '../../assets/channels/E2.svg';
import { ReactComponent as E2GRComponent } from '../../assets/channels/E2GR.svg';
import { ReactComponent as E2NOComponent } from '../../assets/channels/E2NO.svg';
import { ReactComponent as E2RUGComponent } from '../../assets/channels/E2RUG.svg';

export const icons = {
  E2GR: {
    altText: 'Eurosport 2',
    widthRatio: 3.5,
    component: E2GRComponent,
  },
  E2NO: {
    altText: 'Eurosport 2',
    widthRatio: 2.06,
    component: E2NOComponent,
  },
  E2RUG: {
    altText: 'Eurosport 2',
    widthRatio: 4,
    component: E2RUGComponent,
  },
  E1: {
    altText: 'Eurosport 1',
    widthRatio: 1.42,
    component: E1Component,
  },
  E2: {
    altText: 'Eurosport 2',
    widthRatio: 1.66,
    component: E2Component,
  },
  E: {
    altText: 'Eurosport',
    widthRatio: 0.87,
    component: EComponent,
  },
};

export default callsign => {
  if (callsign === 'E2GR') {
    return icons.E2GR;
  }

  if (callsign === 'E2NO') {
    return icons.E2NO;
  }

  if (callsign === 'E2RUG') {
    return icons.E2RUG;
  }

  if (callsign.startsWith('E1')) {
    return icons.E1;
  }

  if (callsign.startsWith('E2')) {
    return icons.E2;
  }
  return icons.E;
};
