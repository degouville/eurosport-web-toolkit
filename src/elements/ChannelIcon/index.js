import React from 'react';
import PropTypes from 'prop-types';
import E from '../../assets/channels/E.svg';
import E1 from '../../assets/channels/E1.svg';
import E2 from '../../assets/channels/E2.svg';
import E2GR from '../../assets/channels/E2GR.svg';
import E2NO from '../../assets/channels/E2NO.svg';
import E2RUG from '../../assets/channels/E2RUG.svg';

export const iconMap = {
  E: {
    src: E,
    altText: 'Eurosport',
    widthRatio: 0.87,
  },
  E1: {
    src: E1,
    altText: 'Eurosport 1',
    widthRatio: 1.42,
  },
  E2: {
    src: E2,
    altText: 'Eurosport 2',
    widthRatio: 1.66,
  },
  E2GR: {
    src: E2GR,
    altText: 'Eurosport 2',
    widthRatio: 3.5,
  },
  E2NO: {
    src: E2NO,
    altText: 'Eurosport 2',
    widthRatio: 2.06,
  },
  E2RUG: {
    src: E2RUG,
    altText: 'Eurosport 2',
    widthRatio: 4,
  },
};

const ChannelIcon = ({ type, height, ...props }) => {
  const icon = iconMap[type] || iconMap.E;

  return icon ? (
    <img
      {...props}
      css={`
        height: ${height}px;
        width: ${icon.widthRatio * height}px;
      `}
      src={icon.src}
      alt={icon.altText}
    />
  ) : null;
};

ChannelIcon.defaultProps = {
  type: 'E',
  height: 50,
};

ChannelIcon.propTypes = {
  type: PropTypes.oneOf(['E', 'E1', 'E2', 'E2NO', 'E2RUG', 'E2GR']),
  height: PropTypes.number,
};

export default ChannelIcon;
