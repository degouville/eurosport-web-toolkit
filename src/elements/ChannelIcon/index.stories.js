import React from 'react';
import { storiesOf } from '@storybook/react';
import { number } from '@storybook/addon-knobs';
import { ChannelIcon } from '../..';

const iconsStories = storiesOf('Elements|ChannelIcon', module);

const iconTypes = ['', 'E1*', 'E2*', 'E2NO', 'E2RUG', 'E2GR'];

iconTypes.forEach(type => {
  iconsStories.add(`ChannelIcon${type ? ` - ${type}` : ''}`, () => (
    <ChannelIcon type={type} height={number('Height', 50)} />
  ));
});
