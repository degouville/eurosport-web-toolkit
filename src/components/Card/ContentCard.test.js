import React from 'react';
import { shallow } from 'enzyme';
import ChannelIcon from 'src/elements/ChannelIcon';
import ContentCard, { StyledPlayIcon, StyledLiveLabel } from './ContentCard';
import CardDetails from './CardDetails';

const cardData = {
  img: 'https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200',
  url: '/article/1',
  category: 'Tennis',
  title: 'Youth olympic summer games',
  description: 'Description',
  timestamp: '09:00 - 10:30',
  channel: 'E1',
  liveLabel: 'live label',
};

it('renders a Article ContentCard', () => {
  const wrapper = shallow(<ContentCard card={cardData} type="article" />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(StyledPlayIcon).length).toEqual(0);
});

it('renders a ContentCard with play icon', () => {
  const wrapper = shallow(<ContentCard card={cardData} type="vod" />);
  expect(wrapper.find(StyledPlayIcon).length).toEqual(1);
});

it('should pass correct props to CardDetails', () => {
  const wrapper = shallow(<ContentCard card={cardData} type="live" />);
  expect(wrapper.find(CardDetails).props()).toEqual({
    card: {
      category: cardData.category,
      title: cardData.title,
      description: cardData.description,
      channel: cardData.channel,
      timestamp: cardData.timestamp,
    },
    icon: <ChannelIcon height={15} type={cardData.channel} />,
  });
});

describe('Live Content Card', () => {
  it('renders a live label', () => {
    const wrapper = shallow(<ContentCard card={cardData} type="live" />);
    expect(
      wrapper
        .find(StyledLiveLabel)
        .render()
        .text()
    ).toContain(cardData.liveLabel);
  });

  it('renders a play icon', () => {
    const wrapper = shallow(<ContentCard card={cardData} type="live" />);
    expect(wrapper.find(StyledPlayIcon).length).toEqual(1);
  });
});
