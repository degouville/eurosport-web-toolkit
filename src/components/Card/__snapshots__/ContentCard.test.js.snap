// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Article ContentCard 1`] = `
<Card>
  <Header>
    <Styled(img)
      alt="Description"
      src="https://i.eurosport.com/2018/10/29/2450727-50913270-2560-1440.jpg?w=200"
    />
  </Header>
  <Content>
    <UppercaseHeading
      as="h2"
    >
      Tennis
    </UppercaseHeading>
    <Styled(Text)>
      Youth olympic summer games
    </Styled(Text)>
    <Styled(Text)>
      Description
    </Styled(Text)>
    <Styled(div)>
      <Styled(Text)>
        09:00 - 10:30
      </Styled(Text)>
    </Styled(div)>
  </Content>
</Card>
`;
